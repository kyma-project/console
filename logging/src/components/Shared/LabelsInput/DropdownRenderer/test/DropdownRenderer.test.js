import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import DropdownRenderer from './../DropdownRenderer';

const mockRecentLabels = ['a', 'b'];

const mockLogLabels = [
  {
    name: 'name 1',
  },
  {
    name: 'name 2',
  },
];

describe('DropdownRenderer', () => {
  it('Renders with minimal props', () => {
    const component = renderer.create(
      <DropdownRenderer
        recentLabels={mockRecentLabels}
        logLabelCategories={mockLogLabels}
        chooseLabel={() => {}}
        loadLabels={() => {}}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Shows "No log labels" text when there is no log labels', () => {
    const component = mount(
      <DropdownRenderer
        recentLabels={mockRecentLabels}
        logLabelCategories={[]}
        chooseLabel={() => {}}
        loadLabels={() => {}}
      />,
    );
    expect(
      component.containsMatchingElement(
        <span data-test-id="no-log-labels">No log labels</span>,
      ),
    ).toBe(true);
  });

  it('Shows "No recent labels" text when there is no recent labels', () => {
    const component = mount(
      <DropdownRenderer
        recentLabels={[]}
        logLabelCategories={mockLogLabels}
        chooseLabel={() => {}}
        loadLabels={() => {}}
      />,
    );
    expect(
      component.containsMatchingElement(
        <span data-test-id="no-recent-labels"> No recent labels </span>,
      ),
    ).toBe(true);
  });
});
