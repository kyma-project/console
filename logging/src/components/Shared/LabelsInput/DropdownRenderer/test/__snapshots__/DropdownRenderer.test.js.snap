// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownRenderer Renders with minimal props 1`] = `
<nav
  className="fd-mega-menu"
>
  <div
    className="fd-mega-menu__group"
  >
    <h3
      className="fd-mega-menu__title"
    >
      Recently Selected Labels
    </h3>
    <ul
      className="fd-mega-menu__list dimmed"
    >
      <li
        className="fd-mega-menu__item"
      >
        <span
          className="cursor-pointer fd-mega-menu__link"
          onClick={[Function]}
        >
          a
        </span>
      </li>
      <li
        className="fd-mega-menu__item"
      >
        <span
          className="cursor-pointer fd-mega-menu__link"
          onClick={[Function]}
        >
          b
        </span>
      </li>
    </ul>
  </div>
  <div
    className="fd-mega-menu__group"
  >
    <h3
      className="fd-mega-menu__title"
    >
      Log Labels
    </h3>
    <ul
      className="fd-mega-menu__list"
    >
      <li
        className="fd-mega-menu__item"
      >
        <span
          aria-controls="name 1"
          aria-haspopup="true"
          aria-label="link"
          className="fd-mega-menu__link has-child cursor-pointer"
          onClick={[Function]}
        >
          name 1
        </span>
        <ul
          aria-hidden={true}
          className="fd-mega-menu__sublist"
          id="name 1"
        >
          <li
            className="fd-mega-menu__subitem"
          >
            <span
              className="fd-mega-menu__sublink"
            >
              No entries
            </span>
          </li>
        </ul>
      </li>
      <li
        className="fd-mega-menu__item"
      >
        <span
          aria-controls="name 2"
          aria-haspopup="true"
          aria-label="link"
          className="fd-mega-menu__link has-child cursor-pointer"
          onClick={[Function]}
        >
          name 2
        </span>
        <ul
          aria-hidden={true}
          className="fd-mega-menu__sublist"
          id="name 2"
        >
          <li
            className="fd-mega-menu__subitem"
          >
            <span
              className="fd-mega-menu__sublink"
            >
              No entries
            </span>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</nav>
`;
